react-app/src/utils/Poller/DataPoller.ts

```diff
@@ -16,14 +16,32 @@ export class DataPoller<TFetcherReturn> {
     return new Promise<{
       maxAttemptsReached: boolean;
       correctDataStateFound: boolean;
-    }>((resolve) => {
+      error?: string;
+    }>((resolve, reject) => {
       let timesPolled = 0;
+      let errorMessage: string | null = null;
 
       const intervalId = setInterval(async () => {
         timesPolled++;
         if (timesPolled <= this.options.pollAttempts) {
-          const data = await this.options.fetcher();
-          const stopPoll = this.options.onPoll(data);
+          errorMessage = null;
+
+          let data: Awaited<TFetcherReturn> | null = null;
+          let stopPoll: boolean = false;
```

Comment:
I prefer the `const` usage. Cognitive-wise, much easier to follow. Could chaining promises produce the desired effect?

Reply:
I'm trying to catch the errors separately so that the error messages are clearer